src/iuse.cpp

```diff
@@ -2117,30 +2125,54 @@ int iuse::sew_advanced(player *p, item *it, bool, const tripoint& )
     rn -= mod_count * 10;                              // Other mods
 
     if( rn <= 8 ) {
-        p->add_msg_if_player(m_bad, _("You damage your %s trying to modify it!"),
-                             mod->tname().c_str());
-        if( mod->inc_damage() ) {
+        p->add_msg_if_player(m_bad, _("You damage your %s trying to modify it and also waste filament (%d)!"),
+            mod->tname().c_str(), batch_size / 2);
+        for( const auto &it : req.get_components() ) {
+            p->consume_items( it, batch_size / 2 );
+        }
+        for( const auto &it : req.get_tools() ) {
+            p->consume_tools( it, batch_size / 2);
+        }
+        if (mod->inc_damage()) {
             p->add_msg_if_player(m_bad, _("You destroy it!"));
-            p->i_rem_keep_contents( pos );
+            p->i_rem_keep_contents(pos);
         }
-        return thread_needed / 2;
+        return 0;
     } else if( rn <= 10 ) {
-        p->add_msg_if_player( m_bad,
-                              _("You fail to modify the clothing, and you waste thread and materials.") );
-        p->consume_items( comps );
-        return thread_needed;
+        p->add_msg_if_player(m_bad,
+            _("You fail to modify the clothing, and you waste filament (%d) and materials."), batch_size);
+        for( const auto &it : req.get_components() ) {
+            p->consume_items( it, batch_size );
+        }
+        for( const auto &it : req.get_tools() ) {
+            p->consume_tools( it, batch_size );
+        }
+        p->consume_items(comps);
+        return 0;
     } else if( rn <= 14 ) {
-        p->add_msg_if_player( m_mixed, _("You modify your %s, but waste a lot of thread."),
-                              mod->tname().c_str() );
-        p->consume_items( comps );
-        mod->item_tags.insert( the_mod );
-        return thread_needed;
+        p->add_msg_if_player(m_mixed, _("You modify your %s, but waste a lot of filament (%d)."), mod->tname().c_str(), batch_size);
+        for( const auto &it : req.get_components() ) {
+            p->consume_items( it, batch_size );
+        }
+        for( const auto &it : req.get_tools() ) {
+            p->consume_tools( it, batch_size );
+        }
+        p->consume_items(comps);
+        mod->item_tags.insert(the_mod);
+        return 0;
     }
 
-    p->add_msg_if_player( m_good, _("You modify your %s!"), mod->tname().c_str() );
-    mod->item_tags.insert( the_mod );
-    p->consume_items( comps );
-    return thread_needed / 2;
+    p->add_msg_if_player(m_good, _("You modify your %s using a total of %d filament!"), mod->tname().c_str(), batch_size);
+    mod->item_tags.insert(the_mod);
+    for( const auto &it : req.get_components() ) {
+        p->consume_items( it, batch_size );
+    }
+    for( const auto &it : req.get_tools() ) {
+        p->consume_tools( it, batch_size );
+    }
+    p->consume_items(comps);
```

Comment:
Hmm... I tested it in-game and it works fine - when I need to consume 16 filament to pad item with leather - 16 threads are consumed.

I will test it again.